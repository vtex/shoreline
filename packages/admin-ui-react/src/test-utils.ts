import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { createSystem } from './system'

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
