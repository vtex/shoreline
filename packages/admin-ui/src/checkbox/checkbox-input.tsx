import { useEffect, useRef } from 'react'
import type { CheckboxOptions as AriakitCheckboxOptions } from 'ariakit/checkbox'
import { Checkbox as AriakitCheckbox } from 'ariakit/checkbox'
import { createComponent, useElement } from '@vtex/admin-ui-react'
import { useForkRef } from '@vtex/admin-ui-hooks'

import * as style from './checkbox.style'

export const CheckboxInput = createComponent<
  typeof AriakitCheckbox,
  CheckboxInputOptions
>((props) => {
  const { error = false, ref: htmlRef, state, ...htmlProps } = props

  const ref = useRef<HTMLInputElement>(null)

  useEffect(() => {
    if (ref.current && state) {
      ref.current.indeterminate = state.value === 'indeterminate'
    }
  }, [state])

  return useElement(AriakitCheckbox, {
    ...htmlProps,
    ref: useForkRef(ref, htmlRef),
    state,
    baseStyle: {
      ...style.checkboxStyle,
      ...(error ? style.error : {}),
    },
  })
})

export interface CheckboxInputOptions extends AriakitCheckboxOptions {
  /**
   * Whether has a error or not
   * @default false
   */
  error?: boolean
  /**
   * Checkbox id
   */
  id?: string
  /**
   * Checkbox input ref
   */
  ref?: React.Ref<HTMLInputElement>
}

export type CheckboxInputProps = React.ComponentPropsWithRef<
  typeof CheckboxInput
>
