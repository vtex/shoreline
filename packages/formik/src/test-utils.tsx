import '@testing-library/jest-dom/extend-expect'
import React, { useRef } from 'react'
import { axe, toHaveNoViolations } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import {
  render as baseRender,
  waitFor,
  screen,
  fireEvent,
} from '@testing-library/react'
import { createSystem } from '@vtex/admin-ui-react'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

expect.extend(toHaveNoViolations)

const [ThemeProvider] = createSystem()

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export {
  render,
  withState,
  axe,
  jestMatchMedia,
  renderHook,
  act,
  waitFor,
  screen,
  fireEvent,
}
