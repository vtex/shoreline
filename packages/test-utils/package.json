{
  "name": "@vtex/shoreline-test-utils",
  "version": "1.0.32",
  "description": "Shoreline test utilities",
  "main": "./dist/index.mjs",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.mts",
  "repository": "vtex/shoreline",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist"
  ],
  "exports": {
    ".": {
      "import": "./dist/index.mjs",
      "types": "./dist/index.d.mts"
    },
    "./setup": {
      "import": "./dist/setup.mjs",
      "types": "./dist/setup.d.mts"
    }
  },
  "engines": {
    "node": ">=20"
  },
  "scripts": {
    "prebuild": "rm -rf dist",
    "dev": "tsup --watch",
    "build": "npm run prebuild && tsup"
  },
  "peerDependencies": {
    "vitest": "0.x"
  },
  "devDependencies": {
    "vitest": "0.34.6"
  },
  "dependencies": {
    "@testing-library/dom": "9.3.4",
    "@testing-library/jest-dom": "6.4.6",
    "@testing-library/react": "14.3.1",
    "@testing-library/user-event": "14.5.2"
  }
}
