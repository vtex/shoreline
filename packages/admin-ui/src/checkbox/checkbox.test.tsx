import React, { Fragment } from 'react'
import { render } from '../../test-utils'

import type { CheckboxStateReturn } from './checkbox-input'
import { Checkbox, useCheckboxState } from './checkbox-input'

function CheckboxState({
  children,
}: {
  children: (state: CheckboxStateReturn) => JSX.Element
}) {
  const state = useCheckboxState()

  return children(state)
}

describe('Checkbox tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <CheckboxState>
        {(state) => (
          <Checkbox
            state={state}
            data-testid="checkbox"
            value="checkbox"
            aria-label="checkbox"
            csx={{ bg: 'azure' }}
          />
        )}
      </CheckboxState>
    )

    expect(getByTestId('checkbox')).toHaveStyleRule('background', 'azure')
  })

  it('should match snapshot with diferent sizes', () => {
    const { asFragment } = render(
      <Fragment>
        <CheckboxState>
          {(state) => (
            <Checkbox
              state={state}
              data-testid="checkbox"
              value="checkbox"
              aria-label="checkbox"
              size="regular"
            />
          )}
        </CheckboxState>
        <CheckboxState>
          {(state) => (
            <Checkbox
              state={state}
              data-testid="checkbox"
              value="checkbox"
              aria-label="checkbox"
              size="small"
            />
          )}
        </CheckboxState>
      </Fragment>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
