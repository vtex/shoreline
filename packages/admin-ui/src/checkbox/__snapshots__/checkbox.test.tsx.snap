// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should match snapshot 1`] = `
<DocumentFragment>
  .emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-7 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-6 >* {
  margin-left: 0.5rem;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 1.25rem;
  height: 1.25rem;
  background: form.control;
  border: 1px solid #CC3D3D;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  cursor: pointer;
  margin: 0rem;
  box-sizing: border-box;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #FBF2F2,0rem 0rem 0rem 0.1875rem #F49494;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:hover:not(:disabled) {
  border: 1px solid #962828;
}

.emotion-0:focus-visible {
  outline: none;
  border: 1px solid #782020;
}

.emotion-0:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked {
  background-color: #2953B2;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-0:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-0:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-0:checked:focus-visible {
  outline: none;
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-0:checked:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23FFFFFF"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-0:checked[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23747474"/></svg>');
}

.emotion-0:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-0:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-0:indeterminate:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23000000"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-0:indeterminate[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23747474"/></svg>');
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-5 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-1 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <input
        aria-checked="false"
        aria-disabled="true"
        class="emotion-0"
        data-command=""
        disabled=""
        id="checkbox"
        style="pointer-events: none;"
        type="checkbox"
      />
      <div
        class="emotion-5"
      >
        <label
          class="emotion-1"
          for="checkbox"
        >
          checkbox
        </label>
        <div
          class="emotion-4"
        >
          <span
            class="emotion-2"
          >
            Help Text
          </span>
          <span
            class="emotion-3"
            role="alert"
          >
            Error text
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Checkbox should match snapshot to checkbox group 1`] = `
<DocumentFragment>
  .emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-7 >*:not(:first-child) {
  margin-top: 0.25rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
}

.emotion-6 >* {
  margin-left: 0.5rem;
}

.emotion-13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 1.25rem;
  height: 1.25rem;
  background: form.control;
  border: 1px solid #CC3D3D;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  cursor: pointer;
  margin: 0rem;
  box-sizing: border-box;
}

.emotion-13:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #FBF2F2,0rem 0rem 0rem 0.1875rem #F49494;
}

.emotion-13:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-13:hover:not(:disabled) {
  border: 1px solid #962828;
}

.emotion-13:focus-visible {
  outline: none;
  border: 1px solid #782020;
}

.emotion-13:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-13:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-13:checked {
  background-color: #2953B2;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-13:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-13:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-13:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-13:checked:focus-visible {
  outline: none;
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-13:checked:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23FFFFFF"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-13:checked[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23747474"/></svg>');
}

.emotion-13:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-13:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-13:indeterminate:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23000000"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-13:indeterminate[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23747474"/></svg>');
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-5 >*:not(:first-child) {
  margin-top: 0.125rem;
}

.emotion-2 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
  cursor: pointer;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
}

.emotion-4 >*:not(:first-child) {
  margin-top: 0rem;
}

.emotion-3 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #545454;
}

.emotion-15 {
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.75rem;
  line-height: 1.3333333333;
  letter-spacing: 0.004em;
  color: #CC3D3D;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  width: 1.25rem;
  height: 1.25rem;
  background: form.control;
  border: 1px solid #B9B9B9;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  border-radius: 4px;
  cursor: pointer;
  margin: 0rem;
  box-sizing: border-box;
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F4F4F4,0rem 0rem 0rem 0.1875rem #B9B9B9;
}

.emotion-1:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-1:hover:not(:disabled) {
  border: 1px solid #545454;
}

.emotion-1:focus-visible {
  outline: none;
  border: 1px solid #545454;
}

.emotion-1:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-1:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}

.emotion-1:checked {
  background-color: #2953B2;
  border: 1px solid #2953B2;
  position: relative;
}

.emotion-1:checked:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.0625rem #F1F4FC,0rem 0rem 0rem 0.1875rem #96B2F2;
}

.emotion-1:checked:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-1:checked:hover {
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-1:checked:focus-visible {
  outline: none;
  background: #20408C;
  border: 1px solid #20408C;
}

.emotion-1:checked:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23FFFFFF"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-1:checked[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M14.0303 3.96993C14.3232 4.26283 14.3232 4.7377 14.0303 5.03059L7.03032 12.0303C6.73743 12.3231 6.26259 12.3232 5.96969 12.0303L2.46969 8.5306C2.17679 8.23772 2.17677 7.76284 2.46965 7.46994C2.76253 7.17703 3.2374 7.17701 3.53031 7.46989L6.49999 10.4393L12.9697 3.9699C13.2626 3.67702 13.7375 3.67703 14.0303 3.96993Z" fill="%23747474"/></svg>');
}

.emotion-1:checked:disabled {
  background: #E6E6E6;
  border: 1px solid #747474;
  color: #747474;
  cursor: not-allowed;
}

.emotion-1:checked:disabled +div label {
  color: #747474;
  cursor: not-allowed;
}



.emotion-1:indeterminate:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23000000"/></svg>');
  position: absolute;
  height: 1rem;
}

.emotion-1:indeterminate[disabled]:after {
  content: url('data:image/svg+xml; utf8, <svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M1.75 8C1.75 7.58579 2.08579 7.25 2.5 7.25H13.5C13.9142 7.25 14.25 7.58579 14.25 8C14.25 8.41421 13.9142 8.75 13.5 8.75H2.5C2.08579 8.75 1.75 8.41421 1.75 8Z" fill="%23747474"/></svg>');
}

.emotion-0 {
  color: #545454;
  font-family: "VTEX Trust",-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'WGHT' 400;
  font-size: 0.875rem;
  line-height: 1.4285714286;
  letter-spacing: 0.002em;
}

.emotion-30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: unset;
  -ms-flex-pack: unset;
  -webkit-justify-content: unset;
  justify-content: unset;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.emotion-30 >*:not(:first-child) {
  margin-top: 1rem;
}

<div
    class="emotion-7"
  >
    <span
      class="emotion-0"
    >
      checkbox group label (optional)
    </span>
    <div
      role="group"
    >
      <div
        class="emotion-30"
      >
        <div
          class="emotion-7"
        >
          <div
            class="emotion-6"
          >
            <input
              aria-checked="false"
              class="emotion-1"
              data-command=""
              id="checkbox1"
              type="checkbox"
            />
            <div
              class="emotion-5"
            >
              <label
                class="emotion-2"
                for="checkbox1"
              >
                checkbox1
              </label>
              <div
                class="emotion-4"
              >
                <span
                  class="emotion-3"
                >
                  Hekp Text
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-6"
          >
            <input
              aria-checked="false"
              class="emotion-1"
              data-command=""
              id="checkbox2"
              type="checkbox"
            />
            <div
              class="emotion-5"
            >
              <label
                class="emotion-2"
                for="checkbox2"
              >
                checkbox2
              </label>
            </div>
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-6"
          >
            <input
              aria-checked="true"
              checked=""
              class="emotion-13"
              data-command=""
              id="checkbox3"
              type="checkbox"
            />
            <div
              class="emotion-5"
            >
              <label
                class="emotion-2"
                for="checkbox3"
              >
                checkbox3
              </label>
              <div
                class="emotion-4"
              >
                <span
                  class="emotion-15"
                  role="alert"
                >
                  Error Text
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-6"
          >
            <input
              aria-checked="false"
              class="emotion-1"
              data-command=""
              id="checkbox4"
              type="checkbox"
            />
            <div
              class="emotion-5"
            >
              <label
                class="emotion-2"
                for="checkbox4"
              >
                checkbox4
              </label>
            </div>
          </div>
        </div>
        <div
          class="emotion-7"
        >
          <div
            class="emotion-6"
          >
            <input
              aria-checked="false"
              class="emotion-1"
              data-command=""
              id="checkbox5"
              type="checkbox"
            />
            <div
              class="emotion-5"
            >
              <label
                class="emotion-2"
                for="checkbox5"
              >
                checkbox5
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
